### PR TITLE
Quiet completion tests

### PR DESCRIPTION
**Pull request by aramprice**
Move code used only in tests out of `completion` package, and use `GinkgoWriter` for STDERR to reduce noise when running specs.

### CODE REVIEW

**Review by a-hassanin — APPROVED** [review]
Looks good!

File: cmd/completion/completion.go [hunk 1]
```diff
@@ -1,176 +1,9 @@
...
-	if err != nil {
-		logger.Debug(logTag, "session initialization (command '%s') error: %v", initCmdName, err)
-	} else {
-		session = NewCmdBridge(cmd, deps).Session()
-	}
-	cmdContext := &CmdContext{}
-	dq := NewDirectorQuery(logger, cmdContext, session)
-	cfMap := NewCompleteFunctionsMap(logger, dq)
-	return NewBoshCompleteWithFunctions(logger, cmdContext, cfMap)
-}
```
[the remark this answers is not in the document]
Comment (aramprice) [reply cmd/completion/completion.go:64]: Unclear. Based on some googling I would have expected to see something like `cmd. GenBashCompletion(...)` based on [this gist](https://gist.github.com/jsteenb2/0973c838b00940ee8ccdf958a79756af) but nothing like this shows in this branch or on `main`

[the remark this answers is not in the document]
Comment (rkoster) [reply cmd/completion/completion.go:64]: @kinjelom do you have more context on this?

[the remark this answers is not in the document]
Comment (kinjelom) [reply cmd/completion/completion.go:64]: @rkoster I still use https://github.com/kinjelom/bosh-cli-completion/
I think this code is missing here: https://github.com/kinjelom/bosh-cli-completion/blob/main/main.go#L22-L29

bosh-cli/main.go:
```
func main() {
	logger := newLogger()
	defer handlePanic()

	ui := boshui.NewConfUI(logger)
	defer ui.Flush()
	if completion.IsItCompletionCommand(os.Args[1:]) {
		// completion support
		blindUi := boshui.NewWrappingConfUI(&completion.BlindUI{}, logger) // only completion can write to stdout
		bc := completion.NewBoshComplete(blindUi, logger)
		if _, err := bc.Execute(os.Args[1:]); err != nil {
			fail(err, ui, logger)
		}
	} else {
		cmdFactory := boshcmd.NewFactory(boshcmd.NewBasicDeps(ui, logger))

		cmd, err := cmdFactory.New(os.Args[1:])
		if err != nil {
			fail(err, ui, logger)
		}

		err = cmd.Execute()
		if err != nil {
			fail(err, ui, logger)
		} else {
			success(ui, logger)
		}
	}
}

```


Should I test it and make a PR over the weekend?

**Review by ystros — APPROVED** [review]

**Pull request merged**